ntbs-service/DataMigration/NotificationMapper.cs

```diff
@@ -141,16 +141,14 @@ private async Task<IEnumerable<IList<Notification>>> GetGroupedResultsAsNotifica
                     .Where(sc => sc.OldNotificationId == id)
                     .Select(AsSocialContextAddress)
                     .ToList();
-                var notificationTransferEvents = await Task.WhenAll(transferEvents
+                var notificationTransferEvents = transferEvents
                     .Where(sc => sc.OldNotificationId == id)
-                    .Select(AsTransferEvent)
-                    .ToList()
-                );
-                var notificationOutcomeEvents = await Task.WhenAll(outcomeEvents
+                    .Select(sc => AsTransferEvent(sc).Result)
```

Comment:
By calling `.Result` you're making the thread block until the task has completed.

It would be better to use a `foreach` with `await` statements inside so that the commands do execute in sequence (avoiding concurrent DB calls) but are also non-blocking

Reply:
You should basically never call `.Result` or `.Wait` on a task